examples/run_lm_finetuning.py

```diff
@@ -527,7 +521,7 @@ def main():
         train_dataset = BERTDataset(args.train_file, tokenizer, seq_len=args.max_seq_length,
                                     corpus_lines=None, on_memory=args.on_memory)
         num_train_steps = int(
-            len(train_dataset) / args.train_batch_size / args.gradient_accumulation_steps * args.num_train_epochs)
+            len(train_dataset) / args.train_batch_size * args.num_train_epochs)
```

Comment:
Actually `num_train_steps` should also be divided by `torch.distributed.get_world_size()` if `args.local_rank != -1` so that this works well in distributed settings. This was fixed in `run_squad.py` (see around line 825 of `run_squad.py`). Could you update it here also?

Reply:
`run_classifier.py` should have the same general formula, do you think you can put them all on the same page since you are fixing this? 😅 Sorry it falls on you to harmonize all these examples with the best practices.

So the general formula should be:
```python
num_train_steps =
            len(train_dataset) // args.train_batch_size // args.gradient_accumulation_steps * args.num_train_epochs
if args.local_rank != -1:
    num_train_steps = num_train_steps // torch.distributed.get_world_size()
```

Reply:
Hmm... are you sure we should divide twice by `args. gradient_accumulation_steps`?
train_batch_size is already getting divided by it on line 414 in run_classifier.py for example:
`args.train_batch_size = int(args.train_batch_size / args.gradient_accumulation_steps)`

Reply:
The original version is correct. (The second time recovers the batch size without considering gradient_accumulation_steps.)

Reply:
Yeah, you guys are right. I reverted the error I made here.

Reply:
@thomwolf should `num_train_steps` be divided by `torch.distributed.get_world_size()` or only `t_total`? Because so far in all examples (except run_lm_finetuning.py) `t_total` already gets divided by `torch.distributed.get_world_size()` if `args.local_rank != -1`. I added that logic to run_lm_finetuning.py in my latest commit as well.